// Copyright 2023 The IREE Authors
//
// Licensed under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

syntax = "proto3";

package keyvaluestore;

// A simple key-value storage service
service KeyValueStore {
  // Provides a value for each key request
  rpc GetValue (GetValueRequest) returns (GetValueResponse) {}
  rpc SetValue (SetValueRequest) returns (SetValueResponse) {}
}

// The request message containing the key
message GetValueRequest {
  bytes key = 1;
}

// The response message containing the value associated with the key
message GetValueResponse {
  bytes value = 1;
}

// The request message to set a value for a key
message SetValueRequest {
  bytes key = 1;
  bytes value = 2;
}

// The response message for SetValueRequest
message SetValueResponse {}

